### PR TITLE
String quant weights type

### PR DESCRIPTION
**Pull request by FusionBolt**
change quant weights type and fix build error for clang

### CODE REVIEW

File: include/nncase/runtime/datatypes.h [hunk 1]
```diff
@@ -38,6 +38,36 @@ typedef enum _datatype : uint8_t
 #undef DEFINE_DATATYPE
 } datatype_t;
 
+inline datatype_t to_datatype_method(std::string_view name)
```
Comment (zhangyang2057) [inline include/nncase/runtime/datatypes.h:41]: How about using datatypes.def and macro to implement it?

File: include/nncase/runtime/datatypes.h [hunk 2]
```diff
@@ -38,6 +38,36 @@ typedef enum _datatype : uint8_t
...
+    else if (name == "float16")
+        return datatype_t::dt_float16;
+    else if (name == "float32")
+        return datatype_t::dt_float32;
+    else if (name == "float64")
+        return datatype_t::dt_float64;
+    else if (name == "bfloat16")
+        return datatype_t::dt_bfloat16;
+    else
+        throw std::runtime_error("Unsupported data type");
```
Comment (zhangyang2057) [inline include/nncase/runtime/datatypes.h:68]: Print the specific unsupported data type.

File: include/nncase/targets/neutral_target.h [hunk 3]
```diff
@@ -27,7 +27,7 @@ class NNCASE_API neutral_target : public target
     void register_target_independent_passes(const module_type_t &type, ir::transforms::pass_manager &pass_mgr) override;
     void register_target_dependent_passes(const module_type_t &type, ir::transforms::pass_manager &pass_mgr, bool use_ptq) override;
     void register_quantize_annotation_passes(const module_type_t &type, ir::transforms::pass_manager &pass_mgr) override;
-    void register_quantize_passes(const module_type_t &type, ir::transforms::pass_manager &pass_mgr, datatype_t quant_type, datatype_t w_quant_type, bool use_mse_quant_w) override;
+    void register_quantize_passes(const module_type_t &type, ir::transforms::pass_manager &pass_mgr, datatype_t quant_type, std::string_view w_quant_type, bool use_mse_quant_w) override;
```
Comment (zhangyang2057) [inline include/nncase/targets/neutral_target.h:30]: It is something odd that quant_type is datatype_t, while w_quant_type is std::string_view

Comment (FusionBolt) [reply include/nncase/targets/neutral_target.h:30]: Only w_quant_type can be int4 or int6 and adding int4 to datatype_t is a wrong design

File: include/nncase/runtime/datatypes.h [hunk 4]
```diff
@@ -38,6 +38,16 @@ typedef enum _datatype : uint8_t
 #undef DEFINE_DATATYPE
 } datatype_t;
 
+inline datatype_t parse_datatype_str([[maybe_unused]] std::string_view name)
+{
+#define DEFINE_DATATYPE(id, t, n, value) \
+    if (name == "" #id "")               \
```
Comment (zhangyang2057) [inline include/nncase/runtime/datatypes.h:44]: strip backspace round #id

**Pull request merged**